File: frameworks/node/express/Implementation/routes/static.ts
import express from "express";

import type { Routes } from "../app.ts";

/** static: express.static, Express's own static-file middleware, serving the payload directory under /static/. */
const files: Routes = (app, p) => {
  // rb:handler static.file
  // rb:wiring static.*
  app.use("/static", express.static(p.directory));
};

export default files;

File: frameworks/go/gorilla-mux/Implementation/static.go
package implementation

import (
	"net/http"

	"github.com/gorilla/mux"
)

// staticRoutes serve the payload directory with net/http's FileServer under a path prefix,
// exactly as mux's README serves static files. mux has no file server of its own.
func staticRoutes(r *mux.Router, p *Payloads) {
	// rb:handler static.file
	r.PathPrefix("/static/").Handler(http.StripPrefix("/static/", http.FileServer(http.Dir(p.Directory))))
}
